### DISCUSSION TITLE
How differentiate between a player mining a block and leftclicking the air with the PlayerHandAnimationEvent

### DISCUSSION THREAD

**Discussion by DrMondGnu** [Minestom/Minestom#1219] [Help]
I don't know how to differentiate between a player just normaly leftclicking and mining a block. When i use the PlayerHandAnimationEvent and the player is mining a block i get an abnormal amount of event triggers which kinda destroys my code.
How can i only listen for normal clicks?

**Comment by GoldenCookieGIT** [comment]
You can listen for the Player Action packet if you want to see when a player start/stops digging a block (https://wiki.vg/Protocol#Player_Action)
